### ISSUE TITLE
`no-classic-classes`: `getSourceModuleName` should only be called on a `CallExpression`, `MemberExpression` or `Identifier`

### ISSUE THREAD

**Issue by bendemboski** [ember-cli/eslint-plugin-ember#1088]
```js
AssertionError [ERR_ASSERTION]: `getSourceModuleName` should only be called on a `CallExpression`, `MemberExpression` or `Identifier`
    at getSourceModuleName (/private/tmp/test-app/node_modules/eslint-plugin-ember/lib/utils/import.js:46:5)
    at /private/tmp/test-app/node_modules/eslint-plugin-ember/lib/utils/import.js:31:49
    at Array.some (<anonymous>)
    at /private/tmp/test-app/node_modules/eslint-plugin-ember/lib/utils/import.js:30:36
    at Array.find (<anonymous>)
    at getSourceModuleNameForIdentifier (/private/tmp/test-app/node_modules/eslint-plugin-ember/lib/utils/import.js:29:6)
    at CallExpression > MemberExpression[property.name="extend"] (/private/tmp/test-app/node_modules/eslint-plugin-ember/lib/rules/no-classic-classes.js:63:37)
    at /private/tmp/test-app/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/private/tmp/test-app/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
```

is triggered by:

```js
import thing from 'place';

export default class Cls {
  init() {
    this.extend();
  }
}
```

(without the import statement, even though it's unused, the error will not happen).

* `eslint@7.20.0`
* `eslint-plugin-ember@10.2.0`
* `ember-source@3.25.1`
* `ember-cli@3.25.0`

Looks similar to #905, but it was fixed by #906 which was released in v8.10.1 and I'm seeing this in 10.2.0.

**Comment by bendemboski** [comment]
Perhaps also noteworthy that it repros with `this.member.chain.extend()` in place of `this.extend()`. My actual use case is the CKEditor schema API:

```javascript
import Plugin from '@ckeditor/ckeditor5-core/src/plugin';

export default class TaggingEditing extends Plugin {
  init() {
    this.editor.model.schema.extend('$text', { allowAttributes: tagAttribute });
    // ...
  }
}
```
